### PR TITLE
W-14324937 : tests running with java17 and runtime version 4.6.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by JuanAller**

### CODE REVIEW

File: src/test/java/org/mule/module/apikit/RoutingTableTestCase.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public class RoutingTableTestCase {
 
   @BeforeClass
   public static void beforeAll() throws IOException {
-    muleContext = mock(MuleContext.class);
+    muleContext = Mockito.mock(MuleContext.class);
```
[the remark this answers is not in the document]
Comment (JuanAller) [reply src/test/java/org/mule/module/apikit/RoutingTableTestCase.java:32]: 👍

File: src/test/java/org/mule/module/apikit/deserializing/AttributeDeserializerTest.java [hunk 2]
```diff
@@ -50,11 +49,11 @@ public class AttributeDeserializerTest {
...
   public static Iterable<Object[]> data() throws Exception {
     ArrayHeaderDelimiter comma = COMMA;
-    ArrayHeaderDelimiter semicolon = createEnumValue(ArrayHeaderDelimiter.class, "SEMICOLON", 2, ";");
     return asList(new Object[][] {
         {comma, new ArrayHeaderAttributeDeserializer(comma)},
-        {comma, new DummyAttributeDeserializer()},
-        {semicolon, new ArrayHeaderAttributeDeserializer(semicolon)}
+        {comma, new DummyAttributeDeserializer()}
+        // {semicolon, new ArrayHeaderAttributeDeserializer(semicolon)} uncomment when semicolon value included in
+        // ArrayHeaderDelimiter
```
Comment (iglosiggio) [inline src/test/java/org/mule/module/apikit/deserializing/AttributeDeserializerTest.java:56]:
```suggestion
```
I don't really understand why we were testing this. If the enum value does not exist then we should get rid of the commented code too.

File: src/test/java/org/mule/module/apikit/RoutingTableTestCase.java [hunk 3]
```diff
@@ -7,14 +7,14 @@
 package org.mule.module.apikit;
 
 import static org.hamcrest.CoreMatchers.hasItems;
-import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
```
Comment (iglosiggio) [inline src/test/java/org/mule/module/apikit/RoutingTableTestCase.java:10]:
```suggestion
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;
```

File: src/test/java/org/mule/module/apikit/RoutingTableTestCase.java [hunk 4]
```diff
@@ -7,14 +7,14 @@
...
 
 import java.io.IOException;
 
 import org.junit.Assert;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.mockito.Mockito;
```
Comment (iglosiggio) [inline src/test/java/org/mule/module/apikit/RoutingTableTestCase.java:17]:
```suggestion
```

File: src/test/java/org/mule/module/apikit/RoutingTableTestCase.java (hunk 1 above)
Comment (iglosiggio) [inline src/test/java/org/mule/module/apikit/RoutingTableTestCase.java:32]:
```suggestion
    muleContext = mock(MuleContext.class);
```

**Review by iglosiggio — APPROVED** [review]

**Review by iglosiggio — DISMISSED** [review]

**Review by dcourtil — APPROVED** [review]

**Review by santiagoGQ — DISMISSED** [review]
LGTM

File: src/test/java/org/mule/module/apikit/MuleVersionUtilsTest.java [hunk 5]
```diff
@@ -7,98 +7,81 @@
...
-import static org.mule.module.apikit.MockingUtils.setAccessible;
 
 public class MuleVersionUtilsTest {
 
   @Test
-  public void newerVersionIsAtLeastTest() throws Exception {
-    setManifestImplementationVersion("4.3.0");
-    assertTrue(MuleVersionUtils.isAtLeast("4.2.0"));
+  public void newerVersionIsAtLeastTest() {
+    isAtLeast("4.3.0", "4.2.0");
```
Comment (dcourtil) [inline src/test/java/org/mule/module/apikit/MuleVersionUtilsTest.java:22]: This is not needed, right?
```suggestion

```

**Review by dcourtil — APPROVED** [review]

**Review by dcourtil — APPROVED** [review]

**Pull request merged**